### ISSUE TITLE
Build fails on FreeBSD with clang-6 compiler

### ISSUE THREAD

**Issue by yurivict** [indygreg/python-zstandard#52]
The FreeBSD package build fails:
```
cc -DNDEBUG -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -fPIC -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/zstd/common -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext -I/usr/local/include/python3.6m -c /wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/decompressor.c -o build/temp.freebsd-11.1-RELEASE-p11-i386-3.6/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/decompressor.o -DZSTD_MULTITHREAD
cc -DNDEBUG -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -fPIC -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/zstd/common -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext -I/usr/local/include/python3.6m -c /wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/compressiondict.c -o build/temp.freebsd-11.1-RELEASE-p11-i386-3.6/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/compressiondict.o -DZSTD_MULTITHREAD
cc -DNDEBUG -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -fPIC -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/zstd/common -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext -I/usr/local/include/python3.6m -c /wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/decompressoriterator.c -o build/temp.freebsd-11.1-RELEASE-p11-i386-3.6/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/decompressoriterator.o -DZSTD_MULTITHREAD
cc -DNDEBUG -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -fPIC -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/zstd/common -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext -I/usr/local/include/python3.6m -c /wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/decompressionreader.c -o build/temp.freebsd-11.1-RELEASE-p11-i386-3.6/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/decompressionreader.o -DZSTD_MULTITHREAD
cc -DNDEBUG -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -fPIC -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/zstd/common -I/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext -I/usr/local/include/python3.6m -c /wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/constants.c -o build/temp.freebsd-11.1-RELEASE-p11-i386-3.6/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/constants.o -DZSTD_MULTITHREAD
/wrkdirs/usr/ports/archivers/py-zstandard/work-py36/zstandard-0.9.1/c-ext/constants.c:82:51: error: use of undeclared identifier 'ZSTD_TARGETLENGTH_MIN'
        PyModule_AddIntConstant(mod, "TARGETLENGTH_MIN", ZSTD_TARGETLENGTH_MIN);
                                                         ^
1 error generated.
not modified: 'build/_zstd_cffi.c'
```

**Comment by indygreg** [comment]
`ZSTD_TARGETLENGTH_MIN` was removed in zstd 1.3.5 then re-added in 1.3.6. As part of porting to 1.3.5, I removed the `TARGETLENGTH_MIN` constant in 51b954ea1d6a39198f43c53a8cdee9edc4cb5732. However, I will likely add it back again as part of porting to 1.3.6. You could cherry-pick that commit to unblock building with FreeBSD for the time-being. Or maybe FreeBSD has already upgraded to zstd 1.3.6 and this issue went away?

**Comment by indygreg** [comment]
... And I added `TARGETLENGTH_MIN` back in because it was re-added by 1.3.6.

The underlying issue here is that python-zstandard needs to be compiled against exactly the version of libzstd that it is developed against. I suppose I could sprinkle `#ifdef` throughout the code to try to handle old libzstd versions. But that is complexity I'd rather not deal with - especially since there will be some distros wanting to support very old versions of libzstd.

But the reality of software packaging/distribution is that *someone* needs to deal with these problems. And I think I'd rather deal with them upstream than have packagers (who aren't as familiar with the code base) making source changes.

If you tell me which version of libzstd you want to be compatible with, I can see about adjusting the source so things "just work." But I likely won't do this for the upcoming 0.10.0 release.
